client/core/core.go

```diff
@@ -2380,6 +2379,12 @@ func (c *Core) DiscoverAccount(dexAddr string, appPW []byte, certI interface{})
 		return nil, false, newError(authErr, "unexpected authDEX error: %v", err)
 	}
 	if dc.acct.isSuspended {
+		// TODO: Recover this account if it has active matches. If there are no
+		// active matches, should we check if the next usable account for this
+		// DEX is also suspended and maybe has active matches?
+		// If we're definite we want to register with a new/different key, clear
+		// the current account key using dc.acct.resetKeys() so subsequent calls
+		// to dc.acct.setupCryptoV2 does not return this suspended key.
 		c.log.Infof("HD account key for %s was reported as suspended. Register to make a new one.", dc.acct.host)
 		return dc.exchangeInfo(), false, nil // just paid=false, no error
```

Comment:
Looks like an edge case we hadn't previously considered. Should suspended accounts that have active matches be recovered to enable the client complete those match negotiations?

Reply:
Thought about that, and if they are recovering that means their db lacks all the match and order data that would be needed to complete those swaps, so I don't see a reason to use it in this case.
The only way to complete those swaps would be to use an existing dexc.db

Reply:
Saying that, `DiscoverAccount` should probably be doing a `c.isRegistered` like `Register` does too.

Reply:
Umm... I'm realizing that I haven't tested recently getting my account suspended.  We want to re-register in that case **without** re-seeding.  Was/is that handled somehow.  The `c.isRegistered` is certainly blocking that from `Register` at least, regardless of the active swaps issue.

Reply:
Here's the gist of what I propose for `DiscoverAccount`, a loop similar to what `registerHD` does.

```go
// Setup our account keys and attempt to authorize with the DEX.
creds := c.creds()

// Start at key index 0 and attempt to authorized accounts until either (1)
// the server indicates the account is not found, and we return paid=false
// to signal a new account should be registered, or (2) an account is found
// that is not suspended, and we return with paid=true after storing the
// discovered account and promoting it to a persistent connection. In this
// process, we will increment key index and try again whenever the connect
// response indicates a suspended account is found. This instance of Core
// lacks and order or match history for this dex to complete any active
// swaps that might exist for a suspended account, so the user had better
// have another instance with this data if they hope to recover those swaps.
var keyIndex uint32
var encKey []byte
for {
	encKey, _, err = dc.acct.setupCryptoV2(creds, crypter, keyIndex)
	if err != nil {
		return dc.exchangeInfo(), false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
	}

	// Discover the account by attempting a 'connect' (authorize) request.
	err = c.authDEX(dc)
	if err != nil {
		var mErr *msgjson.Error
		if errors.As(err, &mErr) && mErr.Code == msgjson.AccountNotFoundError {
			return dc.exchangeInfo(), false, nil // all good, just go register now
		}
		return nil, false, newError(authErr, "unexpected authDEX error: %v", err)
	}
	if dc.acct.isSuspended {
		c.log.Infof("HD account key for %s was reported as suspended. Deriving another account key.", dc.acct.host)
		dc.acct.resetKeys()
		keyIndex++
		continue
	}

	break // great, the account at this key index is paid and ready
}
```

On thing to consider however is helping out any subsequent `Register` that will have to do the same iteration all over if there are suspended accounts, possibly caching `keyIndex` so it can be used to speed up the `Register` call.